src/transformStateWithClones.js

```diff
@@ -8,6 +8,24 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
-}
+  let copyState = { ...state };
+  const arrayStates = [];
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      copyState = {};
+    } else if (action.type === 'addProperties') {
+      copyState = { ...copyState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      copyState = { ...copyState };
```

Comment:
The spread operator is used correctly to create a shallow copy of the state before removing properties.